### PR TITLE
Add mixphp

### PR DESCRIPTION
**Pull request by onanying**

### CODE REVIEW

File: php/mixphp-swoole/bin/swoole.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+require __DIR__ . '/../vendor/autoload.php';
+
+$vega = App\Vega::new();
+$http = new Swoole\Http\Server('0.0.0.0', 3000);
+$http->on('Request', $vega->handler());
+$http->set([
+    'enable_coroutine' => false,
+    'worker_num' => swoole_cpu_num() * 2,
+    'log_file' => '/dev/null',
+    'log_level' => 5,
```
Comment (waghanza) [inline php/mixphp-swoole/bin/swoole.php:11]: As I understand, this redirect logs to `/dev/null`. This way is not interoperable (working and `unix` but not `win`). Is there not a way this work on all systems ?

File: php/mixphp-swoole/composer.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "autoload": {
+    "psr-4": {
+      "App\\": "src/"
+    }
+  },
+  "require": {
+    "mix/vega": "~3.0"
```
Comment (waghanza) [inline php/mixphp-swoole/composer.json:8]:
```suggestion
    "mix/vega": "~3.0.0"
```

`~3.0` could lead to **3.1** 
https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

File: php/mixphp-workerman/bin/workerman.php [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?php
+require __DIR__ . '/../vendor/autoload.php';
+
+$vega = App\Vega::new();
+$http = new Workerman\Worker("http://0.0.0.0:3000");
+$http->onMessage = $vega->handler();
+$http->count = (int)(shell_exec('nproc') ?: 8) * 2;
```
Comment (waghanza) [inline php/mixphp-workerman/bin/workerman.php:7]: seems not consistent with `swoole_cpu_num() * 2` configured for **swoole**

File: php/mixphp-workerman/composer.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "autoload": {
+    "psr-4": {
+      "App\\": "src/"
+    }
+  },
+  "require": {
+    "workerman/workerman": "~4.0.0",
```
Comment (waghanza) [inline php/mixphp-workerman/composer.json:8]:
```suggestion
    "workerman/workerman": "*",
```
we don't care about `workerman` version `here`

File: php/mixphp-workerman/composer.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "mix/vega": "~3.0"
```
Comment (waghanza) [inline php/mixphp-workerman/composer.json:9]:
```suggestion
    "mix/vega": "~3.0.0"
```

File: php/mixphp/composer.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "autoload": {
+    "psr-4": {
+      "App\\": "src/"
+    }
+  },
+  "require": {
+    "mix/vega": "~3.0"
```
Comment (waghanza) [inline php/mixphp/composer.json:8]:
```suggestion
    "mix/vega": "~3.0.0"
```

File: php/mixphp/composer.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "autoload": {
+    "psr-4": {
+      "App\\": "src/"
+    }
```
Comment (waghanza) [inline php/mixphp/composer.json:5]: just for my knowledge : is this not the default / standard in php packages ? I mean what if this is not defined explicitly ?

**Review by waghanza — CHANGES_REQUESTED** [review]
Some changed requested, but thanks for you awesome work ❤️

File: php/mixphp-workerman/bin/workerman.php (hunk 3 above)
Comment (sy-records) [reply php/mixphp-workerman/bin/workerman.php:7]: He is mixphp-workerman here, use `workerman`, not `swoole`

File: php/mixphp/composer.json (hunk 7 above)
Comment (onanying) [reply php/mixphp/composer.json:5]: This is a must https://getcomposer.org/doc/04-schema.md#psr-4

**Review by sy-records — DISMISSED** [review]

**Review by waghanza — CHANGES_REQUESTED** [review]
Could you regenerate ci config ? (or write manually edit `.semaphore/semaphore.yml) ?

To regenerate ci config file : `bundle exec rake ci:config` but you need previously to setup ruby, run `gem install bundler` and run. `bundle` in project directory

**Review by sy-records — APPROVED** [review]

**Comment by waghanza** [comment]
Therree is merge conflicts @onanying

**Comment by onanying** [comment]
> Therree is merge conflicts @onanying

The conflict is a file generated by `bundle exec rake ci:config`, how to deal with it?

**Comment by onanying** [comment]
@waghanza The conflict is resolved, and the new error is not quite clear

https://the-benchmarker.semaphoreci.com/workflows/478d53b7-1d41-4f36-aeca-05ae64f113f0

```
Unprocessable YAML file.
Error: [{"Expected a maximum of 50 items but got 53.", "#/blocks/16/task/jobs"}]
```

**Comment by waghanza** [comment]
We have more than 50 php frameworks. Need to find a solutions, perhaps split php in 2 (for now)

**Comment by waghanza** [comment]
or perhaps, we can exclude `mixphp` from **CI**, but not from results

PS : The root cause will be solved with https://github.com/the-benchmarker/web-frameworks/pull/3958 (the problem is a hard-limit on CI side)

**Comment by onanying** [comment]
> or perhaps, we can exclude `mixphp` from **CI**, but not from results
> 
> PS : The root cause will be solved with #3958 (the problem is a hard-limit on CI side)

`bundle exec rake ci:config` will automatically add `mixphp`

**Comment by waghanza** [comment]
> `bundle exec rake ci:config` will automatically add `mixphp`

yes, but until https://github.com/the-benchmarker/web-frameworks/pull/3958 it could be skipped on certain condition (it keep an eye)

**Comment by onanying** [comment]
> > `bundle exec rake ci:config` will automatically add `mixphp`
> 
> not yet, I will work on that

He seems to be testing all frameworks, not just php  https://the-benchmarker.semaphoreci.com/workflows/094e0179-dcdd-4ba7-a187-b2f58179ad72

**Comment by waghanza** [comment]
> > > `bundle exec rake ci:config` will automatically add `mixphp`
> > 
> > 
> > not yet, I will work on that
> 
> He seems to be testing all frameworks, not just php https://the-benchmarker.semaphoreci.com/workflows/094e0179-dcdd-4ba7-a187-b2f58179ad72

Indeed, since the CI config has been edited, all frameworks are run (I will fix this issue later)

**Review by waghanza — CHANGES_REQUESTED** [review]
CI is OK for `workerman` and `swoole`
- https://the-benchmarker.semaphoreci.com/jobs/aea77e8a-fe71-4589-b4ac-620d292ecf5d
- https://the-benchmarker.semaphoreci.com/jobs/bce03070-2e2e-4467-9951-1b8c3301b3b6
but not for `php-fpm`
- https://the-benchmarker.semaphoreci.com/jobs/7a84cb76-b7b3-43d3-a57c-d765d63b29e9

**Comment by onanying** [comment]
@waghanza CI passed

**Review by sy-records — APPROVED** [review]

**Review by onanying — COMMENTED** [review]
Merged

**Review by waghanza — APPROVED** [review]

**Comment by onanying** [comment]
@waghanza https://web-frameworks-benchmark.netlify.app/result  There is no mixphp content in the latest result, but the latest commit of imi exists

**Comment by waghanza** [comment]
Thanks for checking ... will take a look for the next update ❤️  (friday or saturday)

**Pull request merged**